flow/project.py

```diff
@@ -3305,29 +3414,25 @@ def select(operation):
                 )
                 break
             try:
-                # Change groups to available run _JobOperation(s)
+                # Generate _JobOperation instances for selected groups and aggregates.
                 with self._potentially_buffered():
                     operations = []
                     run_groups = set(self._gather_flow_groups(names))
                     for (
-                        aggregate_store,
-                        aggregate_groups,
-                    ) in self._group_to_aggregator.inverse.items():
-                        matching_groups = set(aggregate_groups) & run_groups
-                        if not matching_groups:
-                            # Skip iteration over aggregate store if no groups match the
-                            # specified names
-                            continue
-                        for aggregate in aggregate_store.values():
-                            for group in matching_groups:
-                                operations.extend(
-                                    group._create_run_job_operations(
-                                        self._entrypoint,
-                                        default_directives,
-                                        aggregate,
-                                        ignore_conditions,
-                                    )
-                                )
+                        aggregate_id,
+                        aggregate,
+                        group,
+                    ) in self._generate_selected_aggregate_groups(
+                        selected_groups=run_groups,
+                    ):
+                        operations.extend(
```

Comment:
```suggestion
                        if aggregate not in aggregates:
                            continue
                        operations.extend(
```

This might save users a little time if they request `run` for a subset of jobs.

Reply:
Good observation - instead of the suggestion you made, I provided the argument `selected_aggregates=aggregates` to the generator. That should end up doing the same thing you suggested, but using the logic in the generator to filter instead.